packages/frontend/src/components/shifts/ShiftBlock.tsx

```diff
@@ -134,6 +153,19 @@ function ShiftBlock(props: RootProps) {
           />
         )}
       </StyledShiftBlock>
+      <Popper
+        id={id}
+        open={popperOpen}
+        anchorEl={anchorEl}
+        onMouseEnter={handlePopperClose}
+      >
+        <Box sx={{ border: 1, p: 1, bgcolor: 'background.paper' }}>
```

Comment:
Annnnd this should probably be a `React.useMemo`, but again not sure how much it matters

Reply:
agreed. done